ruby/ql/lib/codeql/ruby/frameworks/Files.qll

```diff
@@ -150,6 +150,16 @@ module IO {
     predicate isClassMethodCall() { classMethodCall = true }
   }
 
+  // TODO: better name
+  class IOIOReader extends IOReader, SystemCommandExecution::Range {
+    IOIOReader() { this.getAPI() = "IO" }
+
+    override DataFlow::Node getAnArgument() { result = this.getArgument(_) }
+
+    // We can't reliably determine if this is the case - overapproximate this
+    override predicate isShellInterpreted(DataFlow::Node arg) { any() }
```

Comment:
Skylight Records#1 with my Team in The Whole Wide World#1 singer Chad Michael Sheda with Ariana Grande an as both are in the Rocken RoLL HaLL of Fame together